courses/models.py

```diff
@@ -524,3 +526,164 @@ class ProgramEnrollmentAudit(AuditModel):
     @classmethod
     def get_related_field_name(cls):
         return "enrollment"
+
+
+class FinalGradeQuerySet(models.QuerySet):
+    """
+    QuerySet class defining common query parameters for FinalGrade
+    """
+    def passed(self):
+        """
+        Returns a queryset with a filter that will only fetch passed course runs
+        """
+        return self.filter(passed=True)
+
+    def paid_on_edx(self):
+        """
+        Returns a queryset with a filter that will only fetch course runs that are paid on edX.
+        Note that a course run can still be paid via an Order if this value is False for a given user.
+        """
+        return self.filter(course_run_paid_on_edx=True)
+
+    def for_course_run_key(self, edx_course_key):
+        """
+        Returns a queryset with a filter that will only fetch course runs that match a course key
+        """
+        return self.filter(course_run__edx_course_key=edx_course_key)
+
+    def for_course_run_keys(self, edx_course_keys):
+        """
+        Returns a queryset with a filter that will only fetch course runs that match one of many course keys
+        """
+        return self.filter(course_run__edx_course_key__in=edx_course_keys)
+
+
+class FinalGrade(TimestampedModel, AuditableModel):
+    """
+    Model to store edx final grades
+    """
+    # Set a custom manager for this model. This lets us do a couple useful things:
+    # (1) Apply commonly-needed filters by a short, easily-recognized name, and (2) compose a query piece-by-piece.
+    # Docs: https://docs.djangoproject.com/en/1.10/topics/db/managers/#creating-a-manager-with-queryset-methods
+    objects = FinalGradeQuerySet.as_manager()
+
+    user = models.ForeignKey(User, null=False, on_delete=models.CASCADE)
+    course_run = models.ForeignKey(CourseRun, null=False, on_delete=models.CASCADE)
+    grade = models.FloatField(
+        null=True,
+        validators=[MinValueValidator(0.0), MaxValueValidator(1.0)]
+    )
+    passed = models.BooleanField(default=False)
+    status = models.CharField(
+        null=False,
+        choices=[(status, status) for status in FinalGradeStatus.ALL_STATUSES],
+        default=FinalGradeStatus.PENDING,
+        max_length=30,
+    )
+    course_run_paid_on_edx = models.BooleanField(default=False)
```

Comment:
We don't need this `course_run_paid_on_edx`. On xPRO, it's safe to assume that if there is an enrollment, then the courserun was paid for.